Update device quick start

Steve-Mcl opened the pull request:
closes #3526

## Description

Replace the "Quick start" doc for Device Agent with the new _Quick Connect_ (One Time code) method, (but retain the Web UI method for pre-installed device agent situations)

Updated and streamlined the older Quick Start doc with newer images and slimmed down the additional info to point to existing docs (reduce duplication)

Corrected some bad advice in the original Quick Start (running device agent with sudo, not setting directory permissions etc)

## Related Issue(s)

#3526

## Checklist

<!-- https://flowfuse.com/handbook/development/#defining-done -->

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass <!-- If not adding test coverage, please clarify why not? -->
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [x] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

joepavitt commented:
These instructions will walk the the user through the "Manual Setup" - for a "quick start" guide it should just be copy+paste the OTC, and done?

Having a "Manual Setup" section in these instructions should still detail the `--ui` option, but it should no longer be the default.

Additionally, you've still got an explicit step for "Link Device to Application", which is also no longer required given that selection is made when the Device is first registered.

Steve-Mcl commented:
> These instructions will walk the the user through the "Manual Setup" - for a "quick start" guide it should just be copy+paste the OTC, and done?

That was not the intention of the original "Quick start" - i have modified it (and slimmed it down) for the new method.



> Having a "Manual Setup" section in these instructions should still detail the `--ui` option, but it should no longer be the default.

Not sure what you mean.  I DO refer to the original quick start (now moved to a new file and renamed to "Quick Start with Web UI") - which is still relevant for pre-installed agent.


> Additionally, you've still got an explicit step for "Link Device to Application", which is also no longer required given that selection is made when the Device is first registered.

Where are you seeing this?  Both the [new default doc](https://github.com/FlowFuse/flowfuse/pull/3541/files#diff-48c83324e3dd588958daaaab600fc90a8c3fc63b3959dbfe0c4a50ea0b6212f8R22) and the [old UI method](https://github.com/FlowFuse/flowfuse/pull/3541/files#diff-487dc8880f672457e2b179c41be495d54a1d2c923460f5ac7b18faba2cf5795eR21) doc state:
> Fill in the **Device Name** & **Device Type** and select an **Application**

joepavitt commented:
Apologies @Steve-Mcl - I'd pulled the changes locally to check the docs rendering, and for some reason, website didn't update with the new docs you'd written.

joepavitt commented on docs/device-agent/quickstart.md:
```diff
@@ -7,109 +7,59 @@ navOrder: 2
+_NOTE: It is good practice to keep your Device Agent up to date. Update instructions are also provided in the link above._
 
-<img src="images/config_yml2.png" width=500 />
+_NOTE: If you do not have access to a terminal, follow the [Quick Start with Web UI](./quickstart-web-ui.md) guide._
```
_"If you do not have access to a terminal, follow the [Quick Start with Web UI]"_ where the first step is a command line entry?

Steve-Mcl replied:
Yep, I realise that but it is difficult to describe without being overly verbose. 

The intention is that the Web UI can be used with terminal acces OR without (i.e. pre-installed)

joepavitt replied:
Understood, maybe advising that the Web UI allows for more manual control/customisation of the configuration instead, and directing accordingly?

Steve-Mcl replied:
i'll have a go.

Steve-Mcl replied:
Ah, i wonder if you are still seeing the old doc?

I set this as follows (specifically item 2 in the web-ui doc)

> 
> 1. If you have access to a Terminal/Command Prompt and run the following command:
>     ```bash
>     flowfuse-device-agent --ui --ui-user admin --ui-pass password --ui-port 8081
>     ```
>     _You can find more details about the Device Agent flags [here](https://flowfuse.com/docs/device-agent/running/)_
> 2. If the Device Agent is pre-installed on your hardware, follow the manufacturers instructions to start the device agent with the Web UI enabled.

joepavitt replied:
I was referring to the `NOTE: If you do not have access to a terminal, follow the [Quick Start with Web UI](http://localhost:8080/docs/device-agent/quickstart-web-ui) guide.` in the standard `quickstart.md`. Explanation with the `Web UI` doc is fine.

Steve-Mcl replied:
Then i am not sure how else to word this.

I needed a way to link to the alternative setup (the web-ui setup) and this felt like the right place.

I deliberately did not add an entry in the index - i wanted folk to land in this doc (so they can see there is a quicker/default setup method) before venturing off to the web-ui version.

Perhaps we could change the note to:

> NOTE: If you prefer, follow the [Quick Start with Web UI](http://localhost:8080/docs/device-agent/quickstart-web-ui) guide

However, they may decide that is not relevant to them and thus get stuck looking for instructions relevant to a pre-installed env?


Or a bit more verbose:

> NOTE: If you prefer to use the Device Agent WEB UI, follow the [Quick Start with Web UI](http://localhost:8080/docs/device-agent/quickstart-web-ui) guide


Or a bit more explicit:

> NOTE: If your device agent is pre-installed and you do not have command line access, or you prefer to use the Device Agent WEB UI, then you can follow the [Quick Start with Web UI](http://localhost:8080/docs/device-agent/quickstart-web-ui) guide


Please advise.

joepavitt replied:
Option 2

Steve-Mcl replied:
```suggestion
_NOTE: If you prefer to use the Device Agent WEB UI, follow the [Quick Start with Web UI](./quickstart-web-ui.md) guide._
```

@joepavitt to be sure we both agree, I will provide this update as a suggestion. Please apply if correct.

joepavitt commented on docs/device-agent/quickstart-web-ui.md:
The images in this aren't rendering for me, but I can't work out why - I think this is likely a local dev problem, as they're rendering fine in the other `md` file, and they're using the same pathing.

joepavitt commented on docs/device-agent/quickstart.md:
```diff
@@ -7,109 +7,59 @@ navOrder: 2
-### Link Device to Application
-1. Return to the **Devices** section in your FlowFuse application page.
-2. Locate your newly added device and click its menu.
+### Step 2: Connect the Device
```
A need for an explicit pointer to the _"Running Your Agent"_ at the end of this step? This get's it configured, but they can't proceed to "Developer on Device" without having run the agent first.

Steve-Mcl replied:
Done in [99a06b1](https://github.com/FlowFuse/flowfuse/pull/3541/commits/99a06b1a6f8466a58026a6af55f1ae26afbfded8)

joepavitt commented on docs/device-agent/quickstart.md:
````diff
@@ -32,8 +32,12 @@ _NOTE: If you do not have access to a terminal, follow the [Quick Start with Web
     <img src="images/device_cli.png" />
+3. Start the Device Agent using the command instructed by the device agent output e.g:
+   ```bash
+   flowfuse-device=agent -d /opt/flowfuse-device
````
Type here with a `=` instead of `-`, also not sure we need to overcomplicate the command with `-d`, no need to include it, if they're configuring with differences it'll be in the docs accordingly

Steve-Mcl replied:
> also not sure we need to overcomplicate the command with -d, no need to include it, if they're configuring with differences it'll be in the docs accordingly

Actually, I think that was down to my system having the old `flowfuse-device` directory - the OTC is clever enough to let you know that you MUST include that part (I edited the image manually)

I will do a full device add once more, on a clean system and put updated graphic and command in place

Steve-Mcl replied:
Resolved in [7a13d51](https://github.com/FlowFuse/flowfuse/pull/3541/commits/7a13d51575341a8334e0c48ec26c181c45a8d1b1)

knolleary commented:
I assume this can be merged? Been 3 days since it was approved?
